### ISSUE TITLE
[Hunter][Spell] Feign Death not dropping hunter out of combat when the pet is not attacking

### ISSUE THREAD

**Issue by Zarant** [SunwellWoW/Sunwell-TBC-Bugtracker#2005]
Whenever you successfully cast feign death you are supposed to clear threat from all mobs and only drop out of combat if your pet is also out of combat.

~~Currently, whenever you use feign death on a non-boss mob, you drop combat regardless if your pet is attacking something or not~~

EDIT: The issue still persists, but now if you feign you don't drop out of combat regardless if your pet is attacking something or not.

Source:  https://youtu.be/6DTmw7FvKCk?t=1256

EDIT: See the research below, this is most likely an issue with Feign Death, since this particular spell is not supposed to drop you from combat if your pet is not on Stay or Follow.

**Comment by ujellyx** [comment]
To add something important here:
When you set your pet on "passive" and have it not attack the current mobs, that were in combat with you, you should drop combat. Only if your pet is still attacking a target and you use Feign Death you should be kept in combat.

**Comment by Zarant** [comment]
At the clip below you can see the behavior you just described, you can see in his combat log, hunter feigns, his pet gets hit by the enemy pet (Corean hits Angelo for 59.) and then the hunter is put back in combat once he casts scatter shot. So, you should be put back in combat as soon as your pet hits something.
https://youtu.be/h0eVrsYItpI?t=370

**Comment by ujellyx** [comment]
> At the clip below you can see the behavior you just described, you can see in his combat log, hunter feigns, his pet gets hit by the enemy pet (Corean hits Angelo for 59.) and then the hunter is put back in combat once he casts scatter shot. So, you should be put back in combat as soon as your pet hits something.
> https://youtu.be/h0eVrsYItpI?t=370

Yep, exactly what I meant! :)

**Comment by Zarant** [comment]
More research:
https://youtu.be/m8nOaNMHZIQ?t=735
Hunter recalls his pet, start drinking and then sends his pet after someone. You can see his pet spending focus, meaning that the pet is actively attacking someone and shortly after you can notice a health regen tick and then he is put back in combat after hitting an enemy.

Look at this two other videos:

https://youtu.be/2jOKVdoPp7w?t=451 - Hunter uses FD while his pet is chasing after a mage, he doesn't drop out of combat not even for a split second.

https://www.youtube.com/watch?v=A2gKR_ehmwU - This other video, the hunter uses FD multiple times while his pet is attacking other targets, he doesn't drop out of combat not even once.

So, to sum up, spells that drop you out of combat like Feign Death and possibly Invisibility, should only do that if your pet is not attacking anyone.

**Comment by Zarant** [comment]
Adding a vanilla source for clarification: https://youtu.be/XgFHoXedgSc?t=784
At first the hunter's pet is a attacking a target (clearly in combat at this point), and a second later the Hunter recalls his pet with a follow command, uses Feign Death and is immediately put out of combat.

Which is in line with a just described earlier on https://youtu.be/h0eVrsYItpI?t=368 (tbc video)

Right now it seems that if your pet is in combat, regardless if he is attacking something or not you dont drop out of combat. 

Also, editing the title to address the new issue and to avoid any ambiguity.

**Comment by sulowitch** [comment]
only reason why is pet going back to hunter is because of macro. Because if you want to use freezing trap you need to put your pet to passive - he will break your trap with attacks. thats the only reason. the same reason for scater shots.

https://www.reddit.com/r/KronosWoW/comments/8m347h/struggling_with_feign_death_frost_trap_macro/ here you can see macros for vanilla. it is working in TBC the same way.

**Comment by ujellyx** [comment]
This report isn't about the pet behavior tho, but about the fact that the hunter should only drop combat with Feign Death, if the pet doesn't currently attack something actively.

**Comment by ujellyx** [comment]
Last update, that supposedly fixed the issue aka:

> Core | Author: Lukaasm
> 
> Core / Player
> - Hunter won't lose combat due to Feign Death if pet is in combat.

Didn't fix the issue like it should. Now, even if you stop your pet from attacking via petpassive/petfollow you stay in combat, if you use Feign Death. And this should be. You should drop combat, even if your pet is being attacked, as long as it doesn't "actively" attack something else.

@lukaasm

**Comment by Zarant** [comment]
One more clip from TBC: Hunter feigns, start drinking and mobs start chasing his pet, therefore, the pet is clearly in combat but not attacking anything.
https://youtu.be/b1ZZdKla5kc?t=194
